crates/egui/src/widgets/plot/mod.rs

```diff
@@ -1290,18 +1301,30 @@ struct PreparedPlot {
     draw_cursor_x: bool,
     draw_cursor_y: bool,
     draw_cursors: Vec<Cursor>,
+    sharp_grid_lines: bool,
 }
 
 impl PreparedPlot {
     fn ui(self, ui: &mut Ui, response: &Response) -> Vec<Cursor> {
-        let mut shapes = Vec::new();
+        let mut axes_shapes = Vec::new();
 
         for d in 0..2 {
             if self.show_axes[d] {
-                self.paint_axis(ui, d, &mut shapes);
+                self.paint_axis(
+                    ui,
+                    d,
+                    self.show_axes[1 - d],
+                    &mut axes_shapes,
+                    self.sharp_grid_lines,
+                );
             }
         }
 
+        // Sort the axes by contrast so that those with higher contrast are drawn in front.
+        axes_shapes.sort_by(|(_, contrast1), (_, contrast2)| contrast1.total_cmp(contrast2));
```

Comment:
maybe "strength" is a slightly better word?

Reply:
Changed in https://github.com/emilk/egui/pull/2412/commits/848310deeb33cc9b073722cadde41e828bdea450.